[CIVP-10979] Update installed package versions

stephen-hoover opened the pull request:
- python 3.6.0 -> 3.6.1
- awscli 1.11.60 -> 1.11.75
- boto 2.45.0 -> 2.46.1
- boto3 1.4.3 -> 1.4.4
- numpy 1.12.0 -> 1.12.1
- pubnub 4.0.8 -> 4.0.10
- requests 2.12.4 -> 2.13.0
- scipy 0.18.1 -> 0.19.0
- muffnn 1.0.0 -> 1.1.1
- tensorflow 1.0.0 -> 1.1.0

Explicitly added `botocore` v1.5.38. We had `botocore` installed before (it's a dependency of other AWS libraries), but we're now explicitly including the version number.

Commit:
DEP Update installed package versions

- python 3.6.0 -> 3.6.1
- boto 2.45.0 -> 2.46.1
- boto3 1.4.3 -> 1.4.4
- numpy 1.12.0 -> 1.12.1
- pubnub 4.0.8 -> 4.0.10
- requests 2.12.4 -> 2.13.0
- scipy 0.18.1 -> 0.19.0
- muffnn 1.0.0 -> 1.1.1
- tensorflow 1.0.0 -> 1.1.0

beckermr commented on CHANGELOG.md:
```diff
@@ -8,9 +8,22 @@ Version number changes (major.minor.micro) in this package denote the following:
 ## Unreleased
+
+
+## [2.2.0] - 2017-05-02
```
Are you planning to tag tomorrow?

stephen-hoover replied:
Yes, I am, barring objections.

beckermr replied:
None from me. :) I am racing to get to major version 100 before anyone else!

beckermr reviewed (changes requested):
It looks like pip is downgrading botocore from 1.5.38 to 1.5.23 when installing awscli. I hate that I have to figure that out by hand, but fine. In any case, are we cool with this?

Commit:
Make version of botocore consistent

Each increment of `awscli` seems to require a different version of `botocore`. Make the versions consistent between the conda-installed `botocore` and what's required by `awscli`.

stephen-hoover commented:
@beckermr , thanks for catching that version change with `botocore`. I tweaked the `awscli` version so that it matches the conda-installed `botocore` version.

beckermr commented:
Ahh cool. Did you check that it worked still? @mheilman had some issue with all of this that we had to fix and prompted my deep dive into conda+pip.

stephen-hoover commented:
IIRC, the issue that @mheilman ran into had to do with `awscli` forcing the installation of a version of `botocore` which was incompatible with the version of `boto3` we had installed. Everything is consistent now. I've tested that the AWS CLI will run, although I haven't tested any features.

beckermr commented:
Ahhh ok. Good enough for me! LGTM
